File: 134. Gas Station/_134.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        for i,v in enumerate(gas):
            if v-cost[i]>=0:
                if self.isok(gas[i:]+gas[0:], cost[i:]+cost[0:]):
                    return i

        return -1


    def isok(self, gas, cost):
        tmp = 0
        for i,v in enumerate(gas):
            tmp += (v-cost[i])
            if tmp<0:
                return False

        return True

    def canCompleteCircuit1(self, gas: List[int], cost: List[int]) -> int:
        if sum(gas)<sum(cost):
            return -1

        start  = 0
        remain = 0

        for i,v in enumerate(gas):
            remain += (v-cost[i])
            if remain<0:
                start = i+1
                remain = 0
        
        return start

File: 134. Gas Station/134.c
int sum(int* tmp, int size){
    int res = 0;
    for(int i=0; i<size; i++){
        res += *(tmp+i);
    }
    return res;
}

int canCompleteCircuit(int* gas, int gasSize, int* cost, int costSize){
    if (sum(gas, gasSize) < sum(cost, costSize)){
        return -1;
    }

    int start = 0;
    int remain = 0;

    for(int i=0; i<gasSize; i++){
        remain += (*(gas+i) - *(cost+i));
        if(remain<0){
            start = i+1;
            remain = 0;
        }
    }

    return start;
}